gips/0058-indexing-fees.md

```diff
@@ -0,0 +1,218 @@
+
+### Step 4 — Indexing Occurs
+This step is identical to how the Graph functions today.
+
+### Step 5 — Indexer Posts POI
+This step is identical to how the Graph functions today
+
+### Step 6 — Enter Dispute Period
+After the Indexer submits a POI the dispute period begins. Within this period, the consumer has the option to submit a dispute to the arbitrator (specified in step 3). If the arbitrator determines that the terms in step 3 were violated by the Indexer, the Indexer is slashed according to the agreement. The portion of slashed GRT that is returned to the consumer versus the portion that is burned is specified in step 3. During the dispute period, both the consumer’s funds and the Indexer’s funds are frozen in the contract.
+
+### Step 7 — Indexer Retrieves Funds
+After the dispute period, the Indexer retrieves its entitled funds (both collateral and payment) that remain after any slashing events. Of course, because the collateral and slashing provides adequate incentive for Indexers to meet the terms in step 3, step 7 will often consist of the Indexer withdrawing its full collateral and the consumer’s payment for indexing services.
+
+### Step 8 — Continual Syncing and Payment
+Since subgraphs incur additional syncing and continual , the Indexers post the incremental sync gas and disk space required to fully sync the subgraph. After a dispute period and assuming no slashable events, the Indexer can withdraw additional funds as compensation for the additional syncing resources.
+
+### Step 9 - Indexer Retrieves Remaining Funds
+Although a correctness warranty can be reclaimed at the end of a dispute period (e.g. whenever a PoI is submitted + fixed time,) the sync speed warranty can only be reclaimed at the end of the agreement. A PoI may be required periodically to checkpoint syncing, but submitting a PoI and collecting payment for the subgraph synced so far does not obviate the need for the sync speed warranty to continue for the subgraph yet to sync past that checkpoint.
+
+## Desiderata Revisited
+Given the full description of indexing fees in the previous section, it is now possible to show how the new mechanism meets the desiderata:
+
+**Low Uncertainty – Indexer Revenue**
```

Comment:
Would it be worth replacing "low uncertainty" with "high certainty"? That might make it a bit more clear